services/brig/src/Brig/API/User.hs

```diff
@@ -185,15 +185,20 @@ createUser new = do
   -- get invitation and existing account
   (newTeam, teamInvitation, tid) <-
     case newUserTeam new of
-      Just (NewTeamMember i) ->
-        findTeamInvitation (userEmailKey <$> email) i
-          <&> ( \case
-                  Just (inv, info, tid) -> (Nothing, Just (inv, info), Just tid)
-                  Nothing -> (Nothing, Nothing, Nothing)
-              )
-      Just (NewTeamCreator t) -> (Just t,Nothing,) <$> (Just . Id <$> liftIO nextRandom)
-      Just (NewTeamMemberSSO tid) -> pure (Nothing, Nothing, Just tid)
-      Nothing -> return (Nothing, Nothing, Nothing)
+      Just (NewTeamMember i) -> do
```

Comment:
I don't like this change.  I find it more verbose and less readable.

Reply:
It was more for consisentency's sake: In the refactored version every branch ends up in a `pure (x, x, x,)` expression.
Also tuple sections + fmap is harder for me to read than a bind + pure, but I don't feel strongly about.

Reply:
Reverted to previous version in https://github.com/wireapp/wire-server/pull/1670/commits/3a70d46ef17d9bb09d091af17e515950bf1d4e7e